Update STIPS for version 2.0 release (#163)
spacetelescope/STScI-STIPS: docs/installation.rst

Installation
************

STIPS is a simulation tool that depends on other modules such as PSF and exposure time calculators.  These underlying submodules need to be 
installed for STIPS to function properly along with their supporting datasets.  There are multiple options for installation and they are listed 
in this section along with instructions.

STIPS Requirements
##################

* `Pandeia>=1.7`: Exposure time calculator.
* `WebbPSF>=1.0.0`: Nancy Grace Roman PSF calculator. STIPS also requires that `poppy`, a support package used by `webbpsf`, have version `>=1.0.3`
* `astropy`: STIPS uses astropy in order to:

	- Read and write FITS files.
	- Read and write ASCII tables (specifically in the IPAC format).
	- Generate Sersic profile models (if any are in the generated scene).

* `esutil`: Used for retrieving data from sqlite databases in the form of numpy arrays.

>   .. warning::
>      `esutil` is not installed by the stips setup.py because its pip installation has 
>      errors. After installing STIPS, you must run `pip install esutil --no-cache-dir` to
>      have `esutil` available.
>   .. note::
>      `esutil` is only needed if you are using `stips.star_generator`

Reviewer: ```suggestion

.. warning::
   `esutil` is not installed by the stips setup.py because its pip installation has
   errors. After installing STIPS, you must run `pip install esutil --no-cache-dir` to
   have `esutil` available.

.. note::
   `esutil` is only needed if you are using `stips.star_generator`

```
- Tabbing the warning prevented it from rendering properly.
- Warnings and notes require line breaks both above and below.

* `montage_wrapper`: STIPS uses montage to generate mosaics. It is only imported if
  STIPS is asked to generate a multi-detector image.
* `numpy`: STIPS uses numpy extensively for almost everything that it does.
* `photutils`: STIPS uses photutils to determine the flux inside the half-light radius
  in generated Sersic profiles.
* `synphot>=1.1.1` and `stsynphot>=1.1.0`: STIPS uses synphot and stsynphot to generate
  bandpasses, count rates, and zero points. Note that the reference data must

>   also be downloaded, as described below in "Doanloading Required Data".

Reviewer: ```suggestion
  also be downloaded, as described below in :ref:`downloading-required-ref-data`.
```
This, combined with another suggestion I make just above the linked section, allows you to directly link to another section in the same page. The text that appears here will be the title of that section.

* `scipy`: STIPS uses scipy to manipulate its internal images (zoom and rotate).

Finally, STIPS requires a set of data files whose location is marked by setting the environment variable `stips_data`, which will be installed as 
part of these instructions.

> Installing Using Conda and Source

Reviewer: ```suggestion
Installing Using Conda and Source Code
```

##################################

STIPS can be installed using the source code and a Conda environment file.

> If you do not have anaconda or miniconda installed, please visit the `anaconda docs <https://docs.anaconda.com/anaconda/install/>`_ for 

Reviewer: ```suggestion
If you do not have Anaconda or Miniconda installed, please visit the `Anaconda docs <https://docs.anaconda.com/anaconda/install/>`_ for 
```
Consistency in capitalization.

installation instructions.  We have included a Conda environment file for easily installing or updating Conda packages to meet STIPS 
requirements.  Please follow the steps below to install STIPS:

Installing
**********

> 1. You will need to clone the STIPS source code from the `spacetelescope/STScI-STIPS <https://github.com/spacetelescope/STScI-STIPS.git>`_ repository.  `cd` into the directory you would like to store the source code and run::

Reviewer: I recommend replacing the "Installing" section with the following:
```
#. You will need to clone the STIPS source code from the `spacetelescope/STScI-STIPS <https://github.com/spacetelescope/STScI-STIPS.git>`_ repository.  `cd` into the directory you would like to store the source code and run::

    git clone https://github.com/spacetelescope/STScI-STIPS.git

    cd STScI-STIPS

#. The environment file can be used in two ways:

   * To create a new Conda environment named ``stips``::

        conda env create -f environment.yml
        conda activate stips

   * Or, to install to or update an existing and currently active Conda environment::

        conda env update --file environment.yml

#. You can now install STIPS using the cloned source code as follows::

    python setup.py install
```
- Using `#.` automates the numbering process for the outer list.
- Swapping the inner list from numbered to bulleted helps demonstrate that the two options are separate and not to be done in sequence.
- More use of two carats on each side of one to get monospaced text.

Author: Removed the third bullet (on running `python setup.py install`) because the environment file installs STIPS itself.

Installation
************

STIPS is a simulation tool that depends on other modules such as PSF and exposure time calculators.  These underlying submodules need to be installed for STIPS to function properly along with their supporting datasets.  There are multiple options for installation and they are listed in this section along with instructions.

> STIPS Requirements

Author: Done.

    python setup.py install

> Downloading Required Reference Data
> ************************************

Reviewer: ```suggestion
.. _downloading-required-ref-data:

Downloading Required Reference Data
************************************
```
This, combined with another suggestion I made earlier in the document, allows you to directly link to this section  from other locations in this page.

STIPS, Pandeia, and WebbPSF need the reference datasets.
You will need to download the data and add them to your environmental path

> 1. Add the following paths to your bash environmental path. It is recommended that you add the path to your `.bash_profile` file::

Reviewer: ```suggestion
1. Add the following paths to your bash environmental path. It is recommended that you add the path to your ``.bash_profile`` file::
```

		export stips_data="<absolute_path_to_this_folder>/ref_data/stips_data"
		export WEBBPSF_PATH="<absolute_path_to_this_folder>/ref_data/webbpsf-data"
		export PYSYN_CDBS="<absolute_path_to_this_folder>/ref_data/grp/redcat/trds"
		export pandeia_refdata="<absolute_path_to_this_folder>/ref_data/pandeia_data-x.x.x_roman"

Make sure that you have the correct version number for `pandeia_refdata` (replace the "x.x.x").

> 2. `cd` into the "ref_data" directory in your STScI-STIPS clone

Reviewer: ```suggestion
2. ``cd`` into the "ref_data" directory in your ``STScI-STIPS`` clone
```
Monospacing STScI-STIPS here matters since it's a reference to the GitHub repository name and not just the package in general.

> 3. Run the following code (ensuring your stips environment is active)::

Reviewer: ```suggestion
3. Run the following code (ensuring your ``stips`` Conda environment is active)::
```
Monospacing `stips` here makes it clearer that it's referencing the Conda environment created earlier and not just the package in general.

    >>> import stips

> 		    print(stips.__env__report__)

Reviewer: ```suggestion
    >>> print(stips.__env__report__)
```
The added line of Python is missing the prompt and is not on the same indentation level as the existing one.

You should receive an output of the following form::

		STIPS Version x.y.z with Data Version x.y.z at /Some/Path/To/stips_data

		STIPS Grid Generated with x.y.z

		Pandeia version a.b.c with Data Version a.b.c. at /Some/Path/To/pandeia_refdata

> 		Webbpsf Version d.e.f with Data Version d.e.f at /Some/Path/To/webbpsf_data_path

Reviewer: The output block should also render as plain text instead of as Python code. Please replace this code snippet by copy/pasting the following over it:
```
You should receive an output of the following form:

.. code-block:: text

  STIPS Version x.y.z with Data Version x.y.z at /Some/Path/To/stips_data

  STIPS Grid Generated with x.y.z

  Pandeia version a.b.c with Data Version a.b.c. at /Some/Path/To/pandeia_refdata

  Webbpsf Version d.e.f with Data Version d.e.f at /Some/Path/To/webbpsf_data_path
```

> The following warning message can be ignored if it appears::

Reviewer: ```suggestion
Ignore the following warning message if it appears:
```
- Switch to active voice to improve clarity here.
- Only use one colon so the code block that follows does not render as Python code.

>     WARNING: stips_data environment variable not found. Falling back on local STIPS data.

Reviewer: ```suggestion
.. code-block:: text

  WARNING: stips_data environment variable not found. Falling back on local STIPS data.
```
The output block should also render as plain text instead of as Python code.
